sky/utils/rich_utils.py

```diff
@@ -362,14 +362,14 @@ def decode_rich_status(
                     # Replace `\r\n` with `\n`, as printing a line ends with
                     # `\r\n` in linux will cause the line to be empty.
                     line = line[:-2] + '\n'
-                is_payload, line = message_utils.decode_payload(
+                is_payload, decoded_line = message_utils.decode_payload(
                     line, raise_for_mismatch=False)
-                control = None
-                if is_payload:
-                    control, encoded_status = Control.decode(line)
-                if control is None:
+                if not is_payload:
                     yield line
                     continue
+                control, encoded_status = Control.decode(decoded_line)
+                if control is None:
+                    continue
```

Comment:
Existing bug we've always had, where we would print the non-control payload messages, for [example](https://buildkite.com/skypilot-1/smoke-tests/builds/6652/steps/canvas?sid=019b2ee2-79b4-4569-814b-f8465a3602f1#019b2ee2-7bdd-4f64-ac9a-2286fd03af87/L829):

<img width="1395" height="184" alt="Image" src="https://github.com/user-attachments/assets/3338f1fc-7b19-4cfb-b276-ec9c90518d9a" />